### PR TITLE
[Merged by Bors] - feat: Port RingTheory/RingInvo

### PR DESCRIPTION
**Pull request by zaxioms0**
Everything compiles properly, not done with style/removing unnecisary lemmas

### CODE REVIEW

File: Mathlib/RingTheory/RingInvo.lean [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    involution' := involution }
+#align ring_invo.mk' RingInvo.mk'
+
+instance : CoeFun (RingInvo R) fun _ => R → Rᵐᵒᵖ :=
+  ⟨fun f => f.toRingEquiv.toFun⟩
+
+#noalign ring_invo.to_fun_eq_coe
+
+@[simp]
+theorem involution (f : RingInvo R) (x : R) : (f (f x).unop).unop = x :=
```
Comment (mo271) [inline Mathlib/RingTheory/RingInvo.lean:61]:
```suggestion
theorem involution (f : RingInvo R) (x : R) : (f.toRingEquiv  (f.toRingEquiv  x).unop).unop = x :=
```
Maybe this is how to handle the lint error here? I'm really not sure.

File: Mathlib/RingTheory/RingInvo.lean [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+@[norm_cast]
+theorem coe_ring_equiv (f : RingInvo R) (a : R) : (f : R ≃+* Rᵐᵒᵖ) a = f a :=
+  rfl
+#align ring_invo.coe_ring_equiv RingInvo.coe_ring_equiv
+
+@[simp]
+theorem map_eq_zero_iff (f : RingInvo R) {x : R} : f x = 0 ↔ x = 0 :=
+  f.toRingEquiv.map_eq_zero_iff
+#align ring_invo.map_eq_zero_iff RingInvo.map_eq_zero_iff
```
Comment (mo271) [inline Mathlib/RingTheory/RingInvo.lean:77]:
```suggestion
/- Porting note `simp` attribute removed as suggested by linter:
simp can prove this:
  by simp only [AddEquivClass.map_eq_zero_iff]
#noalign ring_invo.map_eq_zero_iff
```
Again I'm not sure how to handle the linter error here correcty. If one replaces `f x` by `f.toRingEquiv` as above, the linter says it is no needed anymore.

**Comment by LukasMias** [comment]
The whole RingInvoClass business needs to be backported before we can merge this PR. The mathlib3 PR for this is found [here](https://github.com/leanprover-community/mathlib/pull/18175).

**Comment by LukasMias** [comment]
> The whole RingInvoClass business needs to be backported before we can merge this PR. The mathlib3 PR for this is found [here](https://github.com/leanprover-community/mathlib/pull/18175). I'm removing the WIP, help, and question tags because I think this PR is finished up to these changes.

The mathlib3 PR is merged now! I will wait to see how mathport translates that file once that happens, and then we can finish things up here :)

**Comment by zaxioms** [comment]
All that's left is to decide whether we want `instance hasCoeToRingEquiv` I'm not sure what the answer here is

File: Mathlib/RingTheory/RingInvo.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/RingTheory/RingInvo.lean:96]: What lint error do you have?

File: Mathlib/RingTheory/RingInvo.lean [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  f.involution' x
+#align ring_invo.involution RingInvo.involution
+
+-- porting note: TODO clarify whether we want a Coe instance like this at all.
+-- instance hasCoeToRingEquiv : Coe (RingInvo R) (R ≃+* Rᵐᵒᵖ) :=
+--   ⟨RingInvo.toRingEquiv⟩
+-- #align ring_invo.has_coe_to_ring_equiv RingInvo.hasCoeToRingEquiv
+
+@[norm_cast]
+theorem coe_ring_equiv (f : RingInvo R) (a : R) : (f : R ≃+* Rᵐᵒᵖ) a = f a :=
```
Comment (jcommelin) [inline Mathlib/RingTheory/RingInvo.lean:106]:
```suggestion
theorem coe_ringEquiv (f : RingInvo R) (a : R) : (f : R ≃+* Rᵐᵒᵖ) a = f a :=
```

File: Mathlib/RingTheory/RingInvo.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/RingTheory/RingInvo.lean:98]: I think there is no issue anymore, right?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**